v3/docs/migration/account-migration.mdx

```diff
@@ -0,0 +1,2284 @@
+                    let supertokensUsersWithSameEmail = await SuperTokens.listUsersByAccountInfo(input.tenantId, {
+                        email: email
+                    }, undefined, input.userContext);
+                    let emailPasswordUser = supertokensUsersWithSameEmail.find(u => {
+                        return u.loginMethods.find(lM => lM.hasSameEmailAs(email) && lM.recipeId === "emailpassword") !== undefined;
+                    })
+
+                    if (emailPasswordUser === undefined) {
+                        // EmailPassword user with the input email does not exist in SuperTokens
+                        // Check if the input credentials are valid in the external provider
+                        let legacyUserInfo = await validateAndGetUserInfoFromExternalProvider(email, password)
+                        if (legacyUserInfo === undefined) {
+                            // credentials are incorrect
+                            return {
+                                status: "WRONG_CREDENTIALS_ERROR"
+                            }
+                        }
+
+                        // Call the signup function to create a new SuperTokens user.
+                        let signUpResponse = await EmailPassword.signUp(input.tenantId, email, password, undefined, input.userContext);
+                        if (signUpResponse.status !== "OK") {
+                            throw new Error("Should never come here")
+                        }
+
+                        // Map the external provider's userId to the SuperTokens userId
+                        await SuperTokens.createUserIdMapping({ superTokensUserId: signUpResponse.user.id, externalUserId: legacyUserInfo.user_id })
+
+                        // Set the userId in the response to use the provider's userId
+                        signUpResponse.user.id = legacyUserInfo.user_id
+                        signUpResponse.user.loginMethods[0].recipeUserId = new RecipeUserId(legacyUserInfo.user_id);
+                        signUpResponse.recipeUserId = new RecipeUserId(legacyUserInfo.user_id);
+
+                        // We will also need to set the email verification status of the user
+                        if (legacyUserInfo.isEmailVerified) {
+                            // Generate an email verification token for the user
+                            let generateEmailVerificationTokenResponse = await EmailVerification.createEmailVerificationToken(input.tenantId, signUpResponse.recipeUserId, email, input.userContext);
+
+                            if (generateEmailVerificationTokenResponse.status === "OK") {
+                                // Verify the user's email
+                                await EmailVerification.verifyEmailUsingToken("public", generateEmailVerificationTokenResponse.token, undefined, input.userContext);
+                            }
+                        }
```

Comment:
I think this part can be replaced by snippet very similar to step 1.1 (Create new users when they sign up using your previous provider)

Reply:
Ping @tamassoltesz for more details on that if necessary

Reply:
I updated the typescript code example to call the bulk migration API. If the sample is ok I'll also change the python and go snippets later.